packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -45,25 +46,53 @@ def draw(self):
             self.figure.dpi = orig_dpi
             self._idle_scheduled = False
 
+    def print_png(self, filename_or_obj, *args,
+                  metadata=None, pil_kwargs=None,
+                  **kwargs):
+
+        from matplotlib import _png
+
+        if metadata is None:
+            metadata = {}
+        metadata = {
+            "Software":
+                f"matplotlib version{__version__}, http://matplotlib.org/",
+            **metadata,
+        }
+
+        canvas = self.get_element('canvas')
+        ctx = canvas.getContext("2d")
+        canvas_data = ctx.getImageData(0, 0, int(ctx.width),
+                                       int(ctx.height)).data
+        data = np.reshape(np.uint8(np.asarray(canvas_data)),
+                                  (int(ctx.height), int(ctx.width), -1))
```

Comment:
Let's use `4` rather than `-1` here.  That way if the data is a different size than we expect it will fail rather than potentially producing an incorrect image.

Reply:
I am unsure about this. This is because the function `_png.write_png` expects a numpy array of shape (M,N,D) where D is the Depth. Now, according to the docs here: https://github.com/matplotlib/matplotlib/blob/db2193ccadeaf022ff0fc4706a9c03db471a8e06/src/_png.cpp#L80

D = 1 represents a grayscale image
D = 3 represents an RGB image and
D = 4 represents an RGBA image

By explicitly putting `-1` as `4`, we would be only catering to the third case.

So, shouldn't we use `-1` then?

Reply:
Yes, but `getImageData` should [always return RGBA data](https://developer.mozilla.org/en-US/docs/Web/API/ImageData#Properties), so if the length was different that would indicate a problem in the width/height usage or somewhere else that it would be good to know about rather than silently do the wrong thing.

Reply:
Sure, Thanks, will make those changes :D